tests/test_salesforce_all_fields_non_custom.py

```diff
@@ -0,0 +1,47 @@
+"""
+Test that with only non-custom fields selected for a stream automatic fields and non custom fields  are still replicated
+"""
+
+from tap_tester.base_suite_tests.all_fields_test import AllFieldsTest
+from sfbase import SFBaseTest
+
+
+class SFNonCustomFieldsTest(AllFieldsTest, SFBaseTest):
+    """Test that with only non-custom fields selected for a stream automatic fields and non custom fields are still replicated
+    TODO
+    This test coveres only BULK api and just couple of streams. We have separate cards to cover these cases
+    TDL-23653: [tap-salesforce]: QA - Add all the streams for the all_fields test
+    TDL-23654: [tap-salesforce]: QA - Add all-fields testcase for REST API streams
+    """
+
+    salesforce_api = 'BULK'
+
+    @staticmethod
+    def name():
+        return "tt_sf_all_fields_non_custom"
+
+    def streams_to_test(self):
+        streams = {'Account', 'Contact'}
+        return streams
+
+    @staticmethod
+    def streams_to_selected_fields():
+        return SFBaseTest.non_custom_fields
+
+    def test_non_custom_fields( self ):
+        for stream in self.streams_to_selected_fields():
+            expected_non_custom_fields = self.streams_to_selected_fields().get(stream, set() )
+            replicated_non_custom_fields = self.actual_fields.get(stream, set() ).difference(self.expected_automatic_fields(stream))
```

Comment:
We don't want to subtract automatic fields here because the automatic fields are non custom fields.

Reply:
The reason to subtract automatic fields - if the replicated fields contain only automatic fields, my assertion for Not none in the next line will not verify if fields other than automatic fields are replicated or not.

Reply:
The automatic fields are non custom fields.  The test is currently making a sub-set of non_custom fields.  Do we need to track both groups?  All non custom fields and then all non custom fields that aren't automatic?  I'm not sure we need to do this.  We could have two assertions if you want to check both cases:
self.assertIsNotNone(replicated_non_custom_fields, msg)
self.assertIsNotNone(replicated_non_custom_fields.difference(self.expected_automatic_fields(stream)), msg)

Reply:
I think this is the test that is still failing.  I suggest not removing automatic fields from replicated_non_custom_fields.  Then we can add two more assertions 

1. The first assertion to make sure that the count of non custom fields is greater than the count of automatic fields to make sure that we are replicating more than just the automatic fields.
2. To make sure the number of non custom fields is equal to the number of replicated fields to make sure we didn't miss any.